### PR TITLE
Add Asciinema Liquid tag

### LINKED ISSUE #4135: Asciinema Liquid tag
**Is your feature request related to a problem? Please describe.**

It would be cool if we can view [Asciinema](https://asciinema.org/) recordings while viewing an article inside dev.to. 

Asciinema does provide a way to embed the video player. 
You can see it [in the respective doc page](https://asciinema.org/docs/embedding). 

**Describe the solution you'd like**

A simple Liquid tag implementation that quickly provides a template for embedding Asciinema recordings. 

The Liquid tag would accept an ID of the recording. 
The ID is basically the number at the end of the URL of the recording (if the link is `https://asciinema.org/a/239367` then the ID is `239367`). 

Using the Liquid tag would look like this: `{% asciinema <ID> %}`.  
Then it would have the equivalent HTML output in the following code listing. 

```
<script id="asciicast-<ID>" src="https://asciinema.org/a/<ID>.js" async></script>
``` 

EDIT as of 2019-09-30: @Glennmen suggested on embedding through an iframe which is a better option, IMO. 
The equivalent HTML output is also simpler. 

```
<iframe src="https://asciinema.org/a/<ID>/iframe"></iframe>
```

### PR DESCRIPTION
**Pull request by citizen428**
Closes #4135

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

Adds a Liquid tag for [Asciinema](https://asciinema.org/) as requested in the issue.

## Related Tickets & Documents

#4135 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed
- [X] added to editor guide

### CODE REVIEW

File: app/views/liquids/_asciinema.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="ltag_asciinema">
+  <iframe frameborder="0"
+    scrolling="no"
+    id="asciinema_<%= id %>"
+    src="https://asciinema.org/a/<%= id %>"
```
Comment (Glennmen) [inline app/views/liquids/_asciinema.html.erb]: I think the `src` is missing `/iframe` at the end.
```suggestion
    src="https://asciinema.org/a/<%= id %>/iframe"
```

**Comment by benhalpern** [comment]
I have no idea how we went this long without asciinema support!

I suspect part of the reason is my thoughts that it could be possible to get this natively integrated so we don't need iframes or anything like that...

https://dev.to/ben/im-very-interested-in-integrating-asciinema-natively-in-devto-cc5

But that is dramatically overthinking this. Awesome to have this PR.

File: app/views/liquids/_asciinema.html.erb (hunk 1 above)
Comment (citizen428) [reply app/views/liquids/_asciinema.html.erb]: Huh, yeah, serves me right for doing this past midnight. Thanks!

**Comment by citizen428** [comment]
@benhalpern Added the missing `/iframe` to the URL now 🙂

File: app/liquid_tags/asciinema_tag.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def parse_id(input)
+    input_no_space = input.delete(" ")
+    raise StandardError, "Invalid Asciinema Id" unless valid_id?(input_no_space)
+
+    input_no_space
+  end
+
+  def valid_id?(id)
+    id.match?(/\A\d+\Z/i)
```
Comment (rhymes) [inline app/liquid_tags/asciinema_tag.rb]: I don't think `/i` is needed since it's only digits, numbers aren't case sensitive :D

Comment (citizen428) [reply app/liquid_tags/asciinema_tag.rb]: Hehe, yeah, this tag was mostly a copy of the Blogcast one, which already had the superfluous `i` modifier I didn't notice.

File: app/liquid_tags/asciinema_tag.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        id: @id
+      },
+    )
+  end
+
+  private
+
+  def parse_id(input)
+    input_no_space = input.delete(" ")
+    raise StandardError, "Invalid Asciinema Id" unless valid_id?(input_no_space)
```
Comment (Zhao-Andy) [inline app/liquid_tags/asciinema_tag.rb]: I'd prefer an error message that showed which ID is erroring out:
```suggestion
    raise StandardError, "Invalid Asciinema ID: {% asciinema #{input_no_space} %}" unless valid_id?(input_no_space)
```

Comment (citizen428) [reply app/liquid_tags/asciinema_tag.rb]: None of the other Liquid tags in `app/liquid_tags` seem to include the ID in the corresponding error, so I opted for consistency. I have no strong feelings though, so if you want it included that's fine.

File: app/views/liquids/_asciinema.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<div class="ltag_asciinema">
+  <iframe frameborder="0"
+    scrolling="no"
+    id="asciinema_<%= id %>"
+    src="https://asciinema.org/a/<%= id %>/iframe"
+    style="width:100%;"
```
[the remark this answers is not in the document]
Comment (citizen428) [reply app/views/liquids/_asciinema.html.erb]: > This is missing a closing </iframe> in the next line.

Which a Cmd-Z brought back 😉 TIL to not submit PRs at 1am when jetlagged :D

File: app/liquid_tags/asciinema_tag.rb (hunk 3 above)
Comment (Zhao-Andy) [reply app/liquid_tags/asciinema_tag.rb]: Yeah, that's more our fault of not improving all the error messages. I think a general trend of making error messages more informative is a good direction.

Comment (citizen428) [reply app/liquid_tags/asciinema_tag.rb]: 👍 done

**Comment by benhalpern** [comment]
This will go live soon. @citizen428 are you interested in making a [changelog](https://dev.to/t/changelog) DEV post?

**Comment by citizen428** [comment]
Currently a bit swamped, so if you want to announce it earlier please go ahead :-)

**Pull request merged**